src/applications/static-pages/download-1095b/components/App/index.js

```diff
@@ -18,20 +18,55 @@ import {
 
 import { CSP_IDS } from '~/platform/user/authentication/constants';
 import { signInServiceName } from '~/platform/user/authentication/selectors';
-import { isLOA1 as isLOA1Selector } from '~/platform/user/selectors';
+import { selectAuthStatus } from '../../selectors/auth-status';
 import {
-  notFoundComponent,
-  unavailableComponent,
   downloadErrorComponent,
   errorTypes,
+  notFoundComponent,
+  systemErrorComponent,
+  unavailableComponent,
 } from './utils';
 import '../../sass/download-1095b.scss';
 
-export const App = ({ displayToggle, isLOA1, loggedIn, toggleLoginModal }) => {
+export const App = ({ displayToggle, toggleLoginModal }) => {
   const [year, updateYear] = useState(0);
   const [formError, updateFormError] = useState({ error: false, type: '' });
   const cspId = useSelector(signInServiceName);
   const [verifyAlertVariant, setverifyAlertVariant] = useState(null);
+  const [availableForms, setAvailableForms] = useState(null);
+  const profile = useSelector(state => selectAuthStatus(state));
+  const isAppLoading = useMemo(
+    () => {
+      return profile.isLoadingProfile || availableForms === null;
+    },
+    [profile, availableForms],
+  );
+
+  useEffect(() => {
+    const getAvailableForms = () => {
+      return apiRequest('/form1095_bs/available_forms')
+        .then(response => {
+          if (response.errors || response.availableForms.length === 0) {
+            updateFormError({ error: true, type: errorTypes.NOT_FOUND });
+          }
+          return response.availableForms === null
+            ? false
+            : response.availableForms;
+        })
+        .catch(() => {
+          updateFormError({ error: true, type: errorTypes.SYSTEM_ERROR });
+          return false; // Return false in case of an error
+        });
+    };
+
+    getAvailableForms().then(forms => {
+      setAvailableForms(forms);
+      const mostRecentYearData = forms[0];
+      if (mostRecentYearData?.lastUpdated && mostRecentYearData?.year) {
+        updateYear(mostRecentYearData.year);
+      }
+    });
+  }, []);
 
```

Comment:
Moved all this from elsewhere in the file and encapsulated in a useEffect that only runs on mount

Reply:
So this runs every time because of the empty dependencies, right? I don't think that's what we want. I think we only want this to run if the user is logged in and loa3 and the feature flag is on. Will this prevent the available forms request from being made unless those conditions are met?

Reply:
I think the dependency still needs to change. We don't want to make the upstream request if the user is not logged in, the feature flag is turned off, or the user isn't loa3. I'm less concerned about the loa3 condition because we will also check that on the backend and return 401 if they're not loa3 (which I think will result in "not found" or "system error" in the current design), but IMO it would be better to guard against that as well.